File: Leetcode_solutions/Reverse Linked List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
  queue = [];
  stock(head, queue);
  reverse(head, queue);
  return head;
};
function stock(h, q) {
  if (h === null) {
    return q;
  }
  q.push(h.val);
  stock(h.next, q);
}
function reverse(h, q) {
  if (h === null) {
    return h;
  }
  h.val = q[q.length - 1];
  q.pop();
  reverse(h.next, q);
}

File: Leetcode_solutions/Reverse_Linked_List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None


class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        prev = None
        curr = head
        while curr != None:
            nextTemp = curr.next
            curr.next = prev
            prev = curr
            curr = nextTemp
        print(prev)
        return prev


class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        queue = []
        self.stock(head, queue)
        self.reverse(head, queue)
        return head

    def stock(self, h, q):
        if h == None:
            return q
        q.append(h.val)
        self.stock(h.next, q)

    def reverse(self, h, q):
        if h == None:
            return h
        h.val = q[len(q) - 1]
        q.pop()
        self.reverse(h.next, q)


class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        prev = None
        curr = head
        while curr != None:
            nextTemp = curr.next
            curr.next = prev
            prev = curr
            curr = nextTemp
        return prev


class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        if head == None:
            return head
        currentHead = head
        while head.next != None:
            p = head.next
            head.next = p.next
            p.next = currentHead
            currentHead = p
        return currentHead


# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        if head == None or head.next == None:
            return head
        memory = []
        dummyHead = ListNode(0)
        dummyHead.next = head
        current = head
        while current != None:
            memory.append(current.val)
            current = current.next
        reversed_memory = memory[::-1]
        i=0
        while head != None:
            head.val = reversed_memory[i]
            i+=1
            head = head.next
        return dummyHead.next